app/controllers/application_controller.rb

```diff
@@ -1,5 +1,12 @@
 class ApplicationController < ActionController::Base
```

Comment:
Adding this code ` before_action :authenticate_user!` to this controller will be great. The reason is, although you have set up sign_up and sign_in correctly and everything is working as expected, your application is misbehaving. For example, a user who is not authenticated can access your application's root page. Also when you sign out a user it doesn't really work as expected. Kindly add that line to your code and test your application and you will notice the difference

Reply:
This is so helpful. Thanks @peterrobert

Reply:
welcome anytime @PeteUgwu  and @bernabasy